pkgs/development/mobile/androidenv/compose-android-packages.nix

```diff
@@ -1,410 +1,707 @@
+          lib.nameValuePair (toVersionKey name) (lib.recurseIntoAttrs (recurse value))
+        else
+          lib.nameValuePair (toVersionKey name) value
+      );
+    in
+    lib.recurseIntoAttrs (recurse liftedArchives);
+
   # Converts a license name to a list of license texts.
   mkLicenses = licenseName: repo.licenses.${licenseName};
 
   # Converts a list of license names to a flattened list of license texts.
   # Just used for displaying licenses.
-  mkLicenseTexts = licenseNames:
-    lib.lists.flatten
-      (builtins.map
-        (licenseName:
-          builtins.map
-            (licenseText: "--- ${licenseName} ---\n${licenseText}")
-            (mkLicenses licenseName))
-      licenseNames);
+  mkLicenseTexts =
+    licenseNames:
+    lib.lists.flatten (
+      builtins.map (
+        licenseName:
+        builtins.map (licenseText: "--- ${licenseName} ---\n${licenseText}") (mkLicenses licenseName)
+      ) licenseNames
+    );
 
   # Converts a license name to a list of license hashes.
-  mkLicenseHashes = licenseName:
-    builtins.map
-      (licenseText: builtins.hashString "sha1" licenseText)
-      (mkLicenses licenseName);
+  mkLicenseHashes =
+    licenseName:
+    builtins.map (licenseText: builtins.hashString "sha1" licenseText) (mkLicenses licenseName);
 
   # The list of all license names we're accepting. Put android-sdk-license there
   # by default.
-  licenseNames = lib.lists.unique ([
-    "android-sdk-license"
-  ] ++ extraLicenses);
-in
-rec {
-  deployAndroidPackages = callPackage ./deploy-androidpackages.nix {
-    inherit stdenv lib mkLicenses;
-  };
-
-  deployAndroidPackage = ({package, buildInputs ? [], patchInstructions ? "", meta ? {}, ...}@args:
-    let
-      extraParams = removeAttrs args [ "package" "os" "buildInputs" "patchInstructions" ];
-    in
-    deployAndroidPackages ({
-      inherit buildInputs;
-      packages = [ package ];
-      patchesInstructions = { "${package.name}" = patchInstructions; };
-    } // extraParams
-  ));
+  licenseNames = lib.lists.unique (
+    [
+      "android-sdk-license"
+    ]
+    ++ extraLicenses
+  );
 
   # put a much nicer error message that includes the available options.
-  check-version = packages: package: version:
+  check-version =
+    packages: package: version:
     if lib.hasAttrByPath [ package version ] packages then
       packages.${package}.${version}
     else
       throw ''
         The version ${version} is missing in package ${package}.
-        The only available versions are ${builtins.concatStringsSep ", " (builtins.attrNames packages.${package})}.
+        The only available versions are ${
+          builtins.concatStringsSep ", " (builtins.attrNames packages.${package})
+        }.
       '';
 
+  # Returns true if we should link the specified plugins.
+  shouldLink =
+    check: packages:
+    assert builtins.isList packages;
+    if check == true then
+      true
+    else if check == false then
```

Comment:
Not sure but shouldn't `check == null` give use the same behavior as `check == false`?

Reply:
I'll probably just remove `null` as an option.